src/main/transaction.c

```diff
@@ -1126,7 +1128,12 @@ queue_pop_wait(queue_t* adata_q)
 	while (1) {
 		adata = queue_pop(adata_q);
 		if (adata == NULL) {
+			#ifdef __aarch64__
```

Comment:
Not sure what is going on here.  Why is this being added?

Reply:
This is the arm64 counterpart to what `_mm_pause()` does. [_mm_pause()](https://android.googlesource.com/platform/prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.7/+/7f61ceb88132c4329380c0cf9634e305dda9fe76/lib/gcc/x86_64-linux-android/4.7/include/xmmintrin.h#1226) decodes to `__asm__ __volatile__ ("rep; nop" : : );` which is the pause instruction on x86. From what I could find it is used to signal to the compiler that we are waiting on a spin lock and some optimizations are possible. I couldn't find a similar function for arm64 so I used the `yield` instruction, which seems to do the same thing. https://developer.arm.com/documentation/dui0802/b/A32-and-T32-Instructions/SEV--SEVL--WFE--WFI--and-YIELD